kustomize/commands/edit/add/addreplacement_test.go

```diff
@@ -0,0 +1,58 @@
+package add
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	testutils_test "sigs.k8s.io/kustomize/kustomize/v5/commands/internal/testutils"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"testing"
+)
+
+const (
+	replaceFileName    = "replacement.yaml"
+	replaceFileContent = `this is just a test file`
+)
+
+func TestAddReplacementWithFilePath(t *testing.T) {
+	fSys := filesys.MakeEmptyDirInMemory()
+	err := fSys.WriteFile(replaceFileName, []byte(replaceFileContent))
+	require.NoError(t, err)
+	testutils_test.WriteTestKustomization(fSys)
+
+	cmd := newCmdAddReplacement(fSys)
+	args := []string{
+		"--path", patchFileName,
+	}
+	cmd.SetArgs(args)
+	assert.NoError(t, cmd.Execute())
+	content, err := testutils_test.ReadTestKustomization(fSys)
+	assert.NoError(t, err)
+	for i := 1; i < len(args); i += 2 {
+		assert.Contains(t, string(content), args[i])
+	}
+}
+
+func TestAddReplacementAlreadyThere(t *testing.T) {
+	fSys := filesys.MakeEmptyDirInMemory()
+	err := fSys.WriteFile(replaceFileName, []byte(replaceFileContent))
+	require.NoError(t, err)
+	testutils_test.WriteTestKustomization(fSys)
```

Comment:
suggestion: since this is verifying that the command will fail if a replacement for the same path is already present, perhaps the kustomization file could be initialized with that content?

Reply:
not really needed. because I add another cmd execute down below so that if you don't write anything in kustomization file and still be fine